### ISSUE TITLE
Entering Text dismisses the view if it contains a Done button

### ISSUE THREAD

**Issue by Supereg** [StanfordBDHG/XCTestExtensions#16]
### Description

Currently, if you enter text into textField using the `enter(value:)` method, the method will press any `Done` button currently contained in the view. This is caused by this line https://github.com/StanfordBDHG/XCTestExtensions/blob/625477e0937294cb3fd6e7bbf72b78f951644b1d/Sources/XCTestExtensions/XCUIElement%2BTextEntry.swift#L212.

### Reproduction

Create a `sheet` that contains a "Done" button in the toolbar, that closes the sheet. Add a text field.
Write UI Tests that use the `enter(value:)` method to enter text into the text field. The sheet will close as the method will arbitrarily search for a "Done" button somewhere.

### Expected behavior

I would expect the "Done" button not to be pressed and instead text to be entered into the textfield.

### Additional context

I don't quite no the rational behind those last few lines in the `selectField(dismissKeyboard:)` method. Would be helpful to get some input to resolve this issue without breaking compatibility.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct and Contributing Guidelines

**Comment by PSchmiedmayer** [comment]
Oh, this is a very good point. this should not be the case! Is there any way that we can detect if these buttons are in the keyboard or limit it to that? Otherwise, we can remove this. The "Continue" button was there for the strange "Swipe to delete tutorial", the done button to also fix an issue where the keyboard was displaying an overlay.

**Comment by Supereg** [comment]
We could replace `app.buttons["Continue"]` with `keyboard.buttons["Continue"]` if we want the "return" button on the bottom right corner. If that was the original intention.

We have a similar pice of code here
https://github.com/StanfordBDHG/XCTestExtensions/blob/625477e0937294cb3fd6e7bbf72b78f951644b1d/Sources/XCTestExtensions/XCUIApplication%2BDismissKeyboard.swift#L14-L22

But that as well isn't complete in the sense that the [SubmitLabel](https://developer.apple.com/documentation/swiftui/submitlabel) can render several different texts.

**Comment by PSchmiedmayer** [comment]
Agree, we would need to cover all the different cases in the `dismissKeyboard` function. 
And regarding the one linked in the issue above: That was mainly due to the swift to type tutorial that show up on the keyboard when you have reset the simulator and show it for a first time, this has also been discussed here: https://developer.apple.com/forums/thread/650826. If we can cover that unfortunate issue in a way that we can only focus on `keyboard.buttons` then I would highly vote for that approach 👍

**Comment by Supereg** [comment]
Just found, the original reported issue is not just a problem with custom placed "Done" buttons, but also if the submit label itself places a "Done" button on the keyboard itself.

I'm not even sure, if the above linked code to handle the "swift to type" tutorial view works as expected. The "Space" seems to be not hittable in any case, as the keyboard is only shown later on(?). Not even sure, I have to experiment a lot more.

**Comment by Supereg** [comment]
@PSchmiedmayer Thanks for the linked discussion. Would be worth linking to it from the source code 👍

**Comment by PSchmiedmayer** [comment]
Agree, let’s do that 👍
